admission/handler_test.go

```diff
@@ -0,0 +1,201 @@
+package admission_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	admissionv1 "k8s.io/api/admission/v1"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	espadmission "github.com/vshn/espejote/admission"
+	espejotev1alpha1 "github.com/vshn/espejote/api/v1alpha1"
+)
+
+func Test_Handler_AdmissionNotFound(t *testing.T) {
+	t.Parallel()
+
+	c := buildFakeClient(t)
+
+	subject := espadmission.NewHandler(c, "default")
+	require.NotNil(t, subject)
+
+	w := httptest.NewRecorder()
+	req := newAdmissionRequest(t, "test", "default", admissionv1.AdmissionRequest{
+		UID: "test",
+	})
+	subject.ServeHTTP(w, req)
+	res := w.Result()
+	require.Equal(t, http.StatusOK, res.StatusCode)
+
+	require.NotNil(t, res.Body)
+	defer res.Body.Close()
+	var admres admissionv1.AdmissionReview
+	require.NoError(t, json.NewDecoder(res.Body).Decode(&admres))
+	require.NotNil(t, admres.Response)
+	require.NotNil(t, admres.Response.Result)
+	require.Equalf(t, http.StatusOK, int(admres.Response.Result.Code), "should allow if Admission object is not found (might have been deleted) message: %s", admres.Response.Result.Message)
+}
+
+func Test_Handler_Allowed(t *testing.T) {
+	t.Parallel()
+
+	c := buildFakeClient(t, &espejotev1alpha1.Admission{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "default",
+		},
+		Spec: espejotev1alpha1.AdmissionSpec{
+			Template: `
+			local esp = import 'espejote.libsonnet';
+
+			esp.ALPHA.admission.allowed("Nice job!")
+`,
+		},
+	})
+
+	subject := espadmission.NewHandler(c, "default")
+	require.NotNil(t, subject)
+
+	w := httptest.NewRecorder()
+	req := newAdmissionRequest(t, "test", "default", admissionv1.AdmissionRequest{
+		UID: "test",
+	})
+	subject.ServeHTTP(w, req)
+	res := w.Result()
+	require.Equal(t, http.StatusOK, res.StatusCode)
+
+	require.NotNil(t, res.Body)
+	defer res.Body.Close()
+	var admres admissionv1.AdmissionReview
+	require.NoError(t, json.NewDecoder(res.Body).Decode(&admres))
+	require.NotNil(t, admres.Response)
+	require.NotNil(t, admres.Response.Result)
+	require.Equalf(t, http.StatusOK, int(admres.Response.Result.Code), "message: %s", admres.Response.Result.Message)
```

Comment:
Shouldn't this be an assert? I would have expected something more along the lines of the two asserts in the Denied test

```suggestion
	assert.Equal(t, http.StatusOK, int(admres.Response.Result.Code)
	assert.Equal(t, "Nice job!", admres.Response.Result.Message)
```